features/nanostack/sal-stack-nanostack/source/libDHCPv6/libDHCPv6_server.c

```diff
@@ -62,7 +62,7 @@ static void libdhcpv6_address_generate(dhcpv6_gua_server_entry_s *serverInfo, dh
         if (entry->linkType == DHCPV6_DUID_HARDWARE_EUI64_TYPE) {
             memcpy(ptr, entry->linkId, 8);
             *ptr ^= 2;
-        } else if (entry->linkType == DHCPV6_DUID_HARDWARE_EUI64_TYPE) {
+        } else if (entry->linkType == DHCPV6_DUID_HARDWARE_EUI48_TYPE) {
```

Comment:
This will be fixed in the Mbed OS 5.11 by https://github.com/ARMmbed/sal-stack-nanostack-private/commit/6cbe7c52623f3138489cc7fb2ce6b46fcb74eaeb